build-system/tasks/check-types-build-system.js

```diff
@@ -0,0 +1,22 @@
+const { execOrThrow } = require('../common/exec');
+
+/**
+ * Performs type checking on the /build-system directory using TypeScript.
+ * Configuration is defined in /build-system/tsconfig.json.
+ *
+ * @return {Promise<void>}
+ */
+function checkBuildSystemTypes() {
+  return new Promise((resolve, reject) => {
+    try {
+      execOrThrow('npx tsc --project ./build-system/tsconfig.json', 'TypeScript build failed');
```

Comment:
I believe throwing an error to the `gulp` task runner will cause it to return a failure with the error. So I think the entire body of this task can be reduced to this one line. 

In general, a preferable option to `exec`ing a command line is to use the native nodeJS API if one exists.